packages/edit-site/src/components/add-new-template/new-template-part.js

```diff
@@ -18,31 +21,54 @@ import CreateTemplatePartModal from '../create-template-part-modal';
 
 export default function NewTemplatePart( { postType } ) {
 	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const { createSuccessNotice, createErrorNotice } = useDispatch(
+		noticesStore
+	);
 
 	async function createTemplatePart( { title, area } ) {
 		if ( ! title ) {
+			createErrorNotice( __( 'Title is not defined.' ), {
+				type: 'snackbar',
+			} );
 			return;
 		}
 
-		const templatePart = await apiFetch( {
-			path: '/wp/v2/template-parts',
-			method: 'POST',
-			data: {
-				slug: kebabCase( title ),
-				title,
-				content: '',
-				area,
-			},
-		} );
+		try {
+			const templatePart = await apiFetch( {
+				path: '/wp/v2/template-parts',
+				method: 'POST',
+				data: {
+					slug: kebabCase( title ),
+					title,
+					content: '',
+					area,
+				},
+			} );
+
+			createSuccessNotice(
+				sprintf(
+					/* translators: The template part title. */
+					__( '"%s" created! Navigating to the editor…' ),
```

Comment:
Can't decide whether we should automatically navigate the user to the created template or show a "View template" link in the success snackbar. @jameskoster WDYT?

Reply:
We display success notice after "navigating" users to a template in post editor Template Mode.

Reply:
I'm not entirely sure we need a Snackbar on _creation_, but let's try it after sending people to the editor like @Mamaduka suggested before deciding.

Reply:
Since we do a full page navigation when creating a template, it's currently not possible to delay the snackbar after the navigation. We could use query params or `sessionStorage` for this though but not trivial either. Should better leave it to another PR if necessary.

Reply:
I think we should just remove this notice if there's going to be a navigation.

For a lot of screenreader users this they won't ever hear this message because of the navigation. Visually it's probably hard to spot too.

Reply:
Yep, the creation notices should be removed.

Reply:
If we want to keep it, I think we should change the exclamation for a period.

I feel like I'm more used to seeing exclamations in error messages to get a user's attention.

Reply:
Ok, I see it has been removed now, I was on a slightly outdated verison of the branch. 👍